src/ApiStatusApi.ts

```diff
@@ -28,24 +28,29 @@ export default class ApiStatusApi {
     }
 
     public async getApiStatus(): Promise<APIStatus> {
-        if (Date.now() - this.lastUpdate < this.cacheDuration) {
-            return this.cached;
+        try {
+            if (Date.now() - this.lastUpdate < this.cacheDuration) {
+                return this.cached;
+            }
+
+            const resp = await fetch(this.apiUrl, {
+                headers: {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json",
+                },
+                method: "GET",
+            });
+
+            if (resp.status !== 200) {
+                throw new Error(`[ApiStatus] Received status code ${resp.status}`);
+            }
+
+            this.cached = await resp.json();
+            this.lastUpdate = Date.now();
+        } catch (error) {
+            console.error(`Error occurred while making a request to the apistatus api: ${error}`);
```

Comment:
Again, you're swallowing errors here that previously would have caused the promise to reject. Rethrow here?